dbt_platform_helper/commands/environment.py

```diff
@@ -572,50 +525,50 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
 
-    for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+    for index, svc in enumerate(services):
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        user_ip = get_public_ip()
+        allowed_ips = list(allowed_ips) + [user_ip]
+        for ip_index, ip in enumerate(allowed_ips):
             create_header_rule(
                 lb_client,
                 listener_arn,
                 target_group_arn,
                 "X-Forwarded-For",
-                allowed_ips,
+                [ip],
                 "AllowedIps",
-            )
-        else:
-            bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
-            create_header_rule(
-                lb_client,
-                listener_arn,
-                target_group_arn,
-                "Bypass-Key",
-                [bypass_value],
-                "BypassIpFilter",
+                (index + 1) * (ip_index + 100),
```

Comment:
part of me would like to suggest extracting this to a variable for readability and debugging purposes, another part of me knows it's not going to make a massive difference, so it's just a small suggestion